src/zenml/integrations/azure/orchestrators/azureml_orchestrator.py

```diff
@@ -470,6 +490,154 @@ def _wait_for_completion() -> None:
                 wait_for_completion=_wait_for_completion,
             )
 
+    def submit_dynamic_pipeline(
+        self,
+        snapshot: "PipelineSnapshotResponse",
+        stack: "Stack",
+        environment: Dict[str, str],
+        placeholder_run: Optional["PipelineRunResponse"] = None,
+    ) -> Optional[SubmissionResult]:
+        """Submits a dynamic pipeline to the orchestrator.
+
+        Dynamic pipelines are executed as an AzureML command job that runs the
+        dynamic pipeline orchestration container.
+
+        Args:
+            snapshot: The pipeline snapshot to submit.
+            stack: The stack the pipeline will run on.
+            environment: Environment variables to set in the orchestration
+                environment.
+            placeholder_run: An optional placeholder run.
+
+        Raises:
+            RuntimeError: If the snapshot contains a schedule.
+
+        Returns:
+            Optional submission result.
+        """
+        logger.info("Submitting dynamic pipeline to AzureML...")
+
+        if snapshot.schedule:
+            raise RuntimeError(
+                "Scheduling dynamic pipelines is not supported for the "
+                "AzureML orchestrator yet."
+            )
+
+        settings = cast(
+            AzureMLOrchestratorSettings,
+            self.get_settings(snapshot),
+        )
+
+        ml_client = self.get_azureml_client()
+
+        image = self.get_image(snapshot=snapshot)
+        env = Environment(
+            name=f"zenml-{snapshot.pipeline_configuration.name}",
+            image=image,
+        )
+
+        entrypoint_command = (
+            DynamicPipelineEntrypointConfiguration.get_entrypoint_command()
+        )
+        entrypoint_args = (
+            DynamicPipelineEntrypointConfiguration.get_entrypoint_arguments(
+                snapshot_id=snapshot.id,
+                run_id=placeholder_run.id if placeholder_run else None,
+            )
+        )
+
+        compute_target = create_or_get_compute(
+            ml_client, settings, default_compute_name=f"zenml_{self.id}"
+        )
+
+        run_name = get_orchestrator_run_name(
+            pipeline_name=snapshot.pipeline_configuration.name
+        )
+
+        command_job = command(
+            name=run_name,
+            command=" ".join(entrypoint_command + entrypoint_args),
+            environment=env,
+            environment_variables=environment,
+            compute=compute_target,
+            experiment_name=snapshot.pipeline_configuration.name,
+            shm_size=settings.shm_size,
+        )
+
+        job = ml_client.jobs.create_or_update(command_job)
+
+        _wait_for_completion = None
+        if settings.synchronous:
+
+            def _wait_for_completion() -> None:
+                logger.info("Waiting for AzureML job to finish...")
+                ml_client.jobs.stream(job.name)
+
+        return SubmissionResult(
+            metadata=self.compute_metadata(job),
```

Comment:
This will be a different job (CommandJob) than the one created by a static pipeline (PipelineJob), but I assume the metadata computation works for both?